### PR TITLE
if variable not found in grid, skip!

### PR DESCRIPTION
**Pull request by rbe051**
I sometimes have variables that only live on some grids (e.g. mechanical displacement only exist in 3D).
  
I'm not sure if we want to do this change. What do you think?

### CODE REVIEW

File: src/porepy/viz/exporter.py [hunk 1]
```diff
@@ -302,6 +302,8 @@ def _write_vtk(self, data, name, g_VTK):
 
         if data is not None:
             for name_field, values_field in data.items():
+                if values_field is None:
+                    continue
                 dataVTK = ns.numpy_to_vtk(values_field.ravel(order='F'),
```
Comment (rbe051) [inline src/porepy/viz/exporter.py]: Do you see any problem by not write a variable to all dimensions?

One problem I see is that you now will not get an error message if you try to export a non-existing variable

**Comment by keileg** [comment]
Is the variable added to the GridBucket, but is None on some grids? If so, we can add a check if the node_props exists (or something, @alessiofumagalli is morally responsible for function names in that class). I would like to get an error message if I miss on a variable name.

**Comment by rbe051** [comment]
It should work for something like this: 
```python
gb.add_node_props(['pressure'])
exporter.write_vtk(['pressure'])
```
I think it still should give an error if pressure is not added to the data, but I can check tomorrow

**Comment by rbe051** [comment]
As I though: 
```python
gb.meshing.simplex_grid(...)
exporter.write_vtk(['pressure'])
```
gives an error, but 
As I though: 
```python
gb.meshing.simplex_grid(...)
gb.add_node_props(['pressure'])
exporter.write_vtk(['pressure'])
```
works. If there are no further objections to this, I will merge this to develop

**Comment by keileg** [comment]
Go ahead.

tor. 4. jan. 2018 kl. 14:23 skrev rbe051 <notifications@github.com>:

**Pull request merged**